lib/src/layouts/awesome/widgets/awesome_pause_resume_button.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:camerawesome/camerawesome_plugin.dart';
+import 'package:flutter/material.dart';
+
+class AwesomePauseResumeButton extends StatefulWidget {
+  final VideoRecordingCameraState state;
+
+  const AwesomePauseResumeButton({
+    super.key,
+    required this.state,
+  });
+
+  @override
+  State<StatefulWidget> createState() {
+    return _AwesomePauseResumeButtonState();
+  }
+}
+
+class _AwesomePauseResumeButtonState extends State<AwesomePauseResumeButton>
+    with SingleTickerProviderStateMixin {
+  late AnimationController _controller;
+  late Animation<double> _animation;
+
+  @override
+  void initState() {
+    _controller =
+        AnimationController(vsync: this, duration: Duration(milliseconds: 300));
+    _animation = Tween<double>(begin: 0.0, end: 1.0).animate(_controller);
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AwesomeOrientedWidget(
+      child: StreamBuilder<MediaCapture?>(
+        stream: widget.state.captureState$,
+        builder: (_, snapshot) {
+          if (snapshot.data?.isRecordingVideo != true) {
+            return SizedBox();
+          }
+
+          bool recordingPaused = snapshot.data!.videoState == VideoState.paused;
```

Comment:
On pourrait bouger cette logique dans le state.

Méthode startOrPause par ex 

D'une façon général avoir toute la logique dans les states. 
Qu'en pense tu ?

Reply:
La logique peut changer en fonction de ce que les gens veulent faire de CamerAwesome dans leur app, donc je pense pas qu'il faille rendre pause et resume private dans tous les cas.

On peut éventuellement rajouter une méthode startOrPause en plus mais je suis pas certain de la clarté d'une telle méthode qui peut faire deux choses opposées selon un état interne...
En tant qu'utilisateur je me vois mal juste appeler une méthode startOrPause sans mettre à jour une UI ou autre. ça pourrait aussi causer des soucis si les gens gèrent mal le moment où appeler cette méthode: spammer un bouton pause pour mettre pause risque de resume la vidéo par exemple.
Tu penses que ça vaut quand même le coup ?

Reply:
Très bon argument. 
C'est vrai que j'ai un peu oublié le custom à ce moment. 

Lets do this.